selfservice/strategy/password/validator.go

```diff
@@ -107,20 +111,20 @@ func lcsLength(a, b string) int {
 	return greatestLength
 }
 
-func (s *DefaultPasswordValidator) fetch(hpw []byte, apiDNSName string) error {
+func (s *DefaultPasswordValidator) fetch(hpw []byte, apiDNSName string) (int64, error) {
```

Comment:
Returning the count here does not rely on the cache having the value in the recursive run.